occlum/run_test_case.sh

```diff
@@ -0,0 +1,112 @@
+
+#!/bin/bash
+TEST_CASE_DIR=/opt/occlum/gvisor_syscall_tests
+BLOCKLIST_DIR=../blocklist
+NGO_BLOCKLIST_DIR=../ngo_blocklist
+FILTER=""
+NGO_FILTER=""
+VERSION=""
+num1=0
+
+get_blocklist_subtests(){
+        if [ -f $BLOCKLIST_DIR/$testcase ]; then
+                FILTER=$(cat $BLOCKLIST_DIR/$testcase | tr '\n' ':')
+        else
+                FILTER=""
+        fi
+}
+
+get_ngo_blocklist_subtests(){
+	if [ -f $NGO_BLOCKLIST_DIR/$testcase ]; then
+		NGO_FILTER=$(cat $NGO_BLOCKLIST_DIR/$testcase | tr '\n' ':')
+	else
+		NGO_FILTER=""
+	fi
+}
+
+run_testcase(){
+	if [ -e /$TEST_CASE_DIR/$testcase ];then
+		cp $TEST_CASE_DIR/$testcase ./image/bin
+                occlum build
+		if [ -z $subtest ];then 
+			if [[ $VERSION = "ngo" ]];then
+				get_blocklist_subtests $testcase
+				get_ngo_blocklist_subtests $testcase
+			elif [[ $VERSION = "occlum" ]];then
+				get_blocklist_subtests $testcase
+			fi
+			FILTER=$FILTER$NGO_FILTER
+                        occlum run /bin/$testcase --gtest_filter=-$FILTER
+                        FILTER=""
+
+		else
+			occlum run /bin/$testcase --gtest_filter=$subtest
+		fi
+
+	else
+		echo "The test suite does not exist"
+		exit -1
+	fi
+}
+
+
+if [ $# -eq 0 ] ;then
+	echo "Missing parameters, you can try './run_test_case.sh -w occlum -s udp_bind_test -t UdpBindTest/SendtoTest.Sendto/0' "
+	exit -1
+fi
+
+rm -rf occlum_workspace
+occlum new occlum_workspace
+pushd occlum_workspace
+until [ $# -eq 0 ]
+do
+	case "$1" in
+		-s)
+			if [ -z $VERSION ];then
+				echo "Missing arguments, you can use '-w occlum' to run test cases that pass in occlum or '-w ngo' to run test cases that pass in ngo"
+				exit -1
+			fi
+			shift 1
+			num1=`expr $num1 + 1`
+			if [ `expr $numl % 2` -eq 0 ];then
```

Comment:
what is `$numl` ? Please check the spelling. And find out why this error is not met by you.

Reply:
Modification complete